### ISSUE TITLE
[Bug]: can't connect to chrome@91 via cdp starting from puppeteer-core@16.0.0

### ISSUE THREAD

**Issue by DudaGod** [puppeteer/puppeteer#9508]
### Bug expectation

I use chrome@91 in CI with selenoid - https://aerokube.com/selenoid/latest/. It means that I start browser using simple http requests and then connect to this browser using url like - ws://selenoid.example.com:4444/devtools/<session-id>
(docs about this url - https://aerokube.com/selenoid/latest/#_accessing_browser_developer_tools).

It works perfectly with puppeteer-core@15.5.0. But stop working with puppeteer-core@16.0.0.  The ws request just hangs and cannot connect to the browser.

I expected that I can connect to chrome@91 via CDP using latest version of puppeteer-core. Or see information on site which browser version supports by each release.

With chrome@101 everything works fine.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
const puppeteer = require('puppeteer-core');

(async () => {
    const browser = await puppeteer.connect({
        browserWSEndpoint: 'ws://my-ci.com/devtools/SESSION_ID',
        defaultViewport: null
    });
})();
```

### Error string

Connection hangs

### Puppeteer version

19.4.0

### Node version

16.16.0

### Package manager

npm

### Package manager version

8.11.0

### Operating system

macOS

**Comment by OrKoN** [comment]
Puppeteer/puppeteer-core only works with the version of Chromium it bundles. https://pptr.dev/chromium-support

**Comment by OrKoN** [comment]
@DudaGod complete coincidence I guess. Puppeteer depends on CDP and a bunch of browser features. All of these change regularly, bugs get introduced, fixed, new features are developed etc. We only have CI coverage for the version we bundle and it is tested with our test suite. Any other combinations might work or not work depending on the exact differences between the binaries.

**Comment by OrKoN** [comment]
Also you probably should not be using v91 as well as v101 anyway, as those are outdated and not maintained. Automation binaries don't have auto-update so they don't get patched with fixes like the regular browser installations.

**Comment by OrKoN** [comment]
I think v16 is mostly because we implemented auto-attach for targets in https://github.com/puppeteer/puppeteer/releases/tag/v16.0.0 that was an existing CDP API which Puppeteer didn't use before. It's needed for race-free interaction with OOPIFs. While working on it we discovered a bunch of issues in CDP which were fixed. Some of them probably prevent the latest puppeteer core version from working with v91.
